assembly/src/parsers/ast/mod.rs

```diff
@@ -66,6 +69,53 @@ impl ProgramAst {
 
         Ok(ProgramAst { local_procs, body })
     }
+
+    /// Compiles the AST into a program that can be executed on Miden VM.
+    #[allow(unreachable_code, unused_variables)]
+    pub fn compile(
```

Comment:
This probably should be `pub(crate)` as this function should not be exposed publicly - but also, I would probably not put it here. The main reason is that it intermingles parsing and compiling - which I'd like to keep separate. I'm imagining this function will need to call a bunch of other functions which have nothing to to do with parsing and would not be located under `/parsers`.

In general, after this PR, all files under `/parsers` should be deleted except for `parsers/ast`. In fact, contents of `/parsers/ast` should become the new contents of `/parsers` - i.e., `parsers/ast` -> `/parsers`.

The compilation-related code should probably go under its own directory. I don't have a great name suggestion for it yet though.

Reply:
N/A; impl moved to `todo`